exif/exif.go

```diff
@@ -209,92 +210,45 @@ type Exif struct {
 //
 // The error can be inspected with functions such as IsCriticalError
 // to determine whether the returned object might still be usable.
-func Decode(r io.Reader) (*Exif, error) {
+func Decode(r io.ReaderAt) (*Exif, error) {
```

Comment:
I don't think I'm okay with this significant breakage of the API.

Reply:
Would the same principle as in the TIFF decoder (`Decode` and `LazyDecode`) be a satisfying alternative for you?

Reply:
That would certainly help - that would leave the only other API breakage being the removal of the exported Exif.Raw field.  Perhaps we could leave Raw in the struct, but only populate it when a doing a non-lazy decode - that should allow existing API behavior to be mostly preserved.